packages/flutter/lib/src/painting/strut_style.dart

```diff
@@ -281,6 +284,58 @@ class StrutStyle extends Diagnosticable {
        assert(leading == null || leading >= 0),
        assert(package == null || (package != null && (fontFamily != null || fontFamilyFallback != null)));
 
+  /// Builds a StrutStyle that contains values of the equivalent properties in
+  /// the provided [textStyle].
+  ///
+  /// The [textStyle] parameter must not be null.
+  ///
+  /// The named parameters override the [textStyle]'s argument's properties.
+  /// Since TextStyle does not contain [leading] or [forceStrutHeight], these values
+  /// will take on default values (null and false) unless otherwise specified.
+  ///
+  /// If provided, fontSize must be positive and non-zero, leading must be zero or positive.
+  ///
+  /// When [textStyle] has a package and a new [package] is also specified, the entire
+  /// font family fallback list should be redefined since the [textStyle]'s package data
+  /// is inherited by being prepended onto the font family names. If
+  /// [fontFamilyFallback] is meant to be empty, pass an empty list instead of null.
+  /// This prevents the previous package name from being prepended twice.
+  StrutStyle.fromTextStyle(TextStyle textStyle, {
+    String fontFamily,
+    List<String> fontFamilyFallback,
+    double fontSize,
+    double height,
+    this.leading, // TextStyle does not have an equivalent (yet).
+    FontWeight fontWeight,
+    FontStyle fontStyle,
+    this.forceStrutHeight,
+    String debugLabel,
+    String package,
+  }) : assert(textStyle != null),
+       assert(fontSize == null || fontSize > 0),
+       assert(leading == null || leading >= 0),
+       assert(package == null || (package != null && (fontFamily != null || fontFamilyFallback != null))),
+       fontFamily = fontFamily != null ? (package == null ? fontFamily : 'packages/$package/$fontFamily') : textStyle.fontFamily,
+       _fontFamilyFallback = fontFamilyFallback ?? textStyle.fontFamilyFallback,
+       height = height ?? textStyle.height,
+       fontSize = fontSize ?? textStyle.fontSize,
+       fontWeight = fontWeight ?? textStyle.fontWeight,
+       fontStyle = fontStyle ?? textStyle.fontStyle,
+       debugLabel = debugLabel ?? textStyle.debugLabel,
+       _package = package; // the textStyle._package data is embedded in the fontFamily names,
+                           // so we no longer need it.
+
+  /// A [StrutStyle] that will have no impact on the text layout.
+  ///
+  /// Equivalent to having no strut at all. All lines will be laid out according to
+  /// the properties defined in [TextStyle].
+  ///
+  /// Default strut is not the same as [StrutStyle.disabled].
```

Comment:
I believe that you mean StrutStyle.disabled != StrutStyle(). 

This line is a little confusing; maybe it doesn't bear saying at all?

Reply:
Yeah, the distinction is between omitting it and having it inherit vs passing [StrutStyle.disabled]. I guess if they dont disable it, it isnt particularly jarring as it doesnt have any direct impact that would be visible.

Reply:
But yeah, it would be inherently true that there was a difference or else it wouldnt exist (in a well designed API at least). I'll remove it.